plugins/kotlin/j2k/shared/src/org/jetbrains/kotlin/nj2k/conversions/LiftReturnsConversion.kt

```diff
@@ -0,0 +1,58 @@
+// Copyright 2000-2024 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
+
+package org.jetbrains.kotlin.nj2k.conversions
+
+import org.jetbrains.kotlin.analysis.api.KtAnalysisSession
+import org.jetbrains.kotlin.nj2k.NewJ2kConverterContext
+import org.jetbrains.kotlin.nj2k.RecursiveConversion
+import org.jetbrains.kotlin.nj2k.asStatement
+import org.jetbrains.kotlin.nj2k.statements
+import org.jetbrains.kotlin.nj2k.tree.*
+
+/**
+ * A code style conversion (disabled in basic mode) that simplifies conditionals that always end in return statements
+ * by lifting out the return
+ *
+ * This is a J2K equivalent of `LiftReturnOrAssignmentInspection`.
+ */
+class LiftReturnsConversion(context: NewJ2kConverterContext) : RecursiveConversion(context) {
+    override fun isEnabledInBasicMode(): Boolean = false
+
+    context(KtAnalysisSession)
+    override fun applyToElement(element: JKTreeElement): JKTreeElement {
+        return when (element) {
+            is JKIfElseStatement -> {
+                element.hasLiftedReturn = element.liftReturnStatement()
+                // ideally we want to return JKReturnStatement() instead of using the "hasLiftedReturn" printer
```

Comment:
@abelkov Do you have any advice/suggestions to achieve this goal? Right now anytime I try to return `JKReturnStatement(element)` or some variation of this I'm running into "check" issues, or else `element` is not in the correct format

Reply:
Maybe @ermattt can help here as well? You can try debugging the code together and see if "check failed" errors can be avoided somehow.